src/storage/hummock_sdk/src/slice_transform.rs

```diff
@@ -0,0 +1,423 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#![allow(dead_code)]
+use std::collections::HashMap;
+use std::sync::Arc;
+
+use risingwave_common::catalog::ColumnDesc;
+use risingwave_common::types::VIRTUAL_NODE_SIZE;
+use risingwave_common::util::ordered::OrderedRowDeserializer;
+use risingwave_common::util::sort_util::OrderType;
+use risingwave_pb::catalog::Table;
+
+use crate::key::{get_table_id, TABLE_PREFIX_LEN};
+
+trait SliceTransform {
+    fn transform<'a>(&self, full_key: &'a [u8]) -> &'a [u8];
```

Comment:
Please change it to mutable method because we may cache the prefix to avoid deserializing every key.

Reply:
yes, actually i introduce the cache in MultiSliceTransform. Shall let SingleSliceTransform mutable ?

Reply:
with some mirco benchmark and discuss with wallace , we decide not to introduce the prefix_cache this PR